src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,171 @@
 package core.basesyntax;
-
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static class Node<T> {
+        T item;
+        Node<T> next;
+        Node<T> prev;
+
+        public Node(T item, Node<T> next, Node<T> prev) {
+            this.item = item;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
```

Comment:
The `Node<T>` class should be moved to the end of the `MyLinkedList` class as it is an inner class.